src/Internal/Service/PaymentProcessingService.php

```diff
@@ -66,7 +66,7 @@ public function process_payment( int $order_id, bool $manual_capture = false ) {
 
 		$request         = new PaymentRequest( $this->legacy_proxy );
 		$initial_state   = $this->state_factory->create_state( InitialState::class, $context );
-		$completed_state = $initial_state->process( $request );
+		$completed_state = $initial_state->start_processing( $request );
```

Comment:
Seems like here we can immediately return the result of `start_processing`, avoiding a variable. This way there would be no misleading `completed_state`, which might not be completed yet.

Reply:
Addressed here: https://github.com/Automattic/woocommerce-payments/pull/7471/commits/ce20eaf4b3aa50f5d0a85af0d1e2d988cc09df10